components/psst/psst_component_installer.cc

```diff
@@ -0,0 +1,170 @@
+  update_client::InstallerAttributes GetInstallerAttributes() const override;
+
+  static base::NoDestructor<std::string> g_psst_component_id_;
+  static base::NoDestructor<std::string> g_psst_component_base64_public_key_;
+
+ private:
+  const std::string component_id_;
+  const std::string component_name_;
+  uint8_t component_hash_[kHashSize];
+};
+
+base::NoDestructor<std::string>
+    PsstComponentInstallerPolicy::g_psst_component_id_(kPsstComponentId);
+base::NoDestructor<std::string>
+    PsstComponentInstallerPolicy::g_psst_component_base64_public_key_(
+        kPsstComponentBase64PublicKey);
+
+PsstComponentInstallerPolicy::PsstComponentInstallerPolicy()
+    : component_id_(kPsstComponentId), component_name_(kPsstComponentName) {
+  // Generate hash from public key.
+  std::string decoded_public_key;
+  base::Base64Decode(kPsstComponentBase64PublicKey, &decoded_public_key);
+  crypto::SHA256HashString(decoded_public_key, component_hash_, kHashSize);
+}
+
+PsstComponentInstallerPolicy::~PsstComponentInstallerPolicy() = default;
+
+bool PsstComponentInstallerPolicy::SupportsGroupPolicyEnabledComponentUpdates()
+    const {
+  return true;
+}
+
+bool PsstComponentInstallerPolicy::RequiresNetworkEncryption() const {
+  return false;
+}
+
+update_client::CrxInstaller::Result
+PsstComponentInstallerPolicy::OnCustomInstall(
+    const base::Value::Dict& manifest,
+    const base::FilePath& install_dir) {
+  return update_client::CrxInstaller::Result(0);
+}
+
+void PsstComponentInstallerPolicy::OnCustomUninstall() {}
+
+void PsstComponentInstallerPolicy::ComponentReady(const base::Version& version,
+                                                  const base::FilePath& path,
+                                                  base::Value::Dict manifest) {
+  PsstService::GetInstance()->LoadPsstRules(path);
+}
+
+bool PsstComponentInstallerPolicy::VerifyInstallation(
+    const base::Value::Dict& manifest,
+    const base::FilePath& install_dir) const {
+  return true;
+}
+
+base::FilePath PsstComponentInstallerPolicy::GetRelativeInstallDir() const {
+  return base::FilePath::FromUTF8Unsafe(component_id_);
+}
+
+void PsstComponentInstallerPolicy::GetHash(std::vector<uint8_t>* hash) const {
+  hash->assign(component_hash_, component_hash_ + kHashSize);
+}
+
+std::string PsstComponentInstallerPolicy::GetName() const {
+  return component_name_;
+}
+
+update_client::InstallerAttributes
+PsstComponentInstallerPolicy::GetInstallerAttributes() const {
+  return update_client::InstallerAttributes();
+}
+
+void OnRegistered() {
+  BraveOnDemandUpdater::GetInstance()->OnDemandUpdate(kPsstComponentId);
+}
+
+void RegisterPsstComponent(component_updater::ComponentUpdateService* cus) {
+  // In test, |cus| could be nullptr.
+  if (!cus) {
+    return;
+  }
+
+  auto installer = base::MakeRefCounted<component_updater::ComponentInstaller>(
+      std::make_unique<PsstComponentInstallerPolicy>());
+  installer->Register(cus, base::BindOnce(&OnRegistered));
```

Comment:
so rather than have a dependency on BraveOnDemandUpdater here, you can pass the callback into `RegisterPsstComponent` and then pass it through to `installer->Register(cus, std::move(on_registered_callback))`

Reply:
`RegisterPsstComponent` is called by `RegisterComponentsForUpdate()` in chromium_src/chrome/browser/component_updater/registration.cc, you're saying create a new callback there?